modules/programs/msmtp.nix

```diff
@@ -0,0 +1,77 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.msmtp;
+
+  dag = config.lib.dag;
+
+  msmtpAccounts = filter (a: a.msmtp.enable)
+    (attrValues config.accounts.email.accounts);
+
+  onOff = p: if p then "on" else "off";
+
+  accountStr = account: with account;
+    concatStringsSep "\n" (
+      [ "account ${name}" ]
+      ++ mapAttrsToList (n: v: n + " " + v) (
+        {
+          host = smtp.host;
+          from = address;
+          auth = "on";
+          user = userName;
+          tls = onOff smtp.tls.enable;
+          tls_starttls = onOff smtp.tls.useStartTls;
+          tls_trust_file = smtp.tls.certificatesFile;
+        }
+        // optionalAttrs (smtp.port != null) {
+          port = toString smtp.port;
+        }
+        // optionalAttrs (passwordCommand != null) {
+          # msmtp requires the password to finish with a newline.
+          passwordeval = ''${pkgs.bash}/bin/bash -c "${toString passwordCommand}; echo"'';
```

Comment:
I wouldn't force this for 2 reasons:
- msmtp can work without passwordeval, if it's not set it will a try aspecific command via libsecret.
- what if the passwordCommand already includes the newline ?
The newline thing should be added in doc/wiki example (and hopefully changed upstream if there is no good reason).

Reply:
I'm not certain what you mean by force. This code will only include the `passwordeval` line if the `passwordCommand` option is non-`null` so if you don't give a password command it should be left out.

I agree that it would be nice to avoid the Bash thing. So I will make this just include `passwordCommand` as-is and add a note to the description of `passwordCommand` that it should end with a newline.

Reply:
I ended up doing the description change and then immediately regretted it. I don't at all like requiring the terminating newline, I now would prefer require not including the newline 🙂

I wrote a mail to the msmtp maintainer asking whether it would be possible to change the msmtp behavior w.r.t. newline at the end of the output.